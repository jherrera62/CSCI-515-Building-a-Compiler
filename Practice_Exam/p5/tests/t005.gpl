//nested repeat-until statements
int count_down=5;
int count_up=0;
on akey
{

  repeat {
      print("Outer count down: " + count_down);
    count_down--;

    repeat {
      count_up++;
      print("  Count up: " + count_up);
    } until(count_up==3);

    count_up=0;
  } until(count_down==0);

}
