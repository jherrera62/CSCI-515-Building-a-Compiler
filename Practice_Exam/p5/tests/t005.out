gpl.cpp::main()
  input file(tests/t005.gpl)
  random seed(42)
  read_keypresses_from_standard_input(true)
  dump_pixels(false)
  symbol_table(true)
  graphics(true)

gpl.cpp::main() Calling yyparse()


gpl.cpp::main() after call to yyparse().


Printing the symbol table from main()
int count_down = 5
int count_up = 0
gpl.cpp::main() Calling window->initialize().


gpl.cpp::main() Passing control to window->main_loop().
print[8]: Outer count down: 5
print[13]:   Count up: 1
print[13]:   Count up: 2
print[13]:   Count up: 3
print[8]: Outer count down: 4
print[13]:   Count up: 1
print[13]:   Count up: 2
print[13]:   Count up: 3
print[8]: Outer count down: 3
print[13]:   Count up: 1
print[13]:   Count up: 2
print[13]:   Count up: 3
print[8]: Outer count down: 2
print[13]:   Count up: 1
print[13]:   Count up: 2
print[13]:   Count up: 3
print[8]: Outer count down: 1
print[13]:   Count up: 1
print[13]:   Count up: 2
print[13]:   Count up: 3

User quit program. Printing the symbol table.
int count_down = 0
int count_up = 0
